### ISSUE TITLE
Focus management on dialog menus

### ISSUE THREAD

**Issue by amazingphilippe** [cds-snc/notification-planning#778]
## Description

The mobile navigation menus are presented like dialogs, although missing the necessary aria roles. However, it may be more understandable to treat the mobile menu as a disclosure navigation pattern. Some related changes would be to preserve the visibility of the original button as a means to close the menu, and tabbing through the menu would also close it. It may be worthwhile to test out the current and suggested mobile menu implementations with users testing.

https://www.w3.org/TR/wai-aria-practices/examples/disclosure/disclosure-navigation.html

## Acceptance Criteria

- [ ] when you close the menu, the focus should move to the dialog trigger
- [ ] Elements outside of the open dialog should not be focusable

**Comment by yaelberger-commits** [comment]
Blocked due to time and capacity constraints until 2023

**Comment by amazingphilippe** [comment]
Compatibility tests with Fable did not yield any further issues. [Review findings in Dovetail](https://cds-snc.dovetailapp.com/insights/74tQGx4PHlPxx3gi5grRWH)

**Comment by YedidaZalik** [comment]
Not adding to Ally statement because looks like it can be closed

**Comment by YedidaZalik** [comment]
@yaelberger-commits please see the comment from Phil in October 2023, above. Given that, can we close this file?
